build-system/tasks/helpers.js

```diff
@@ -529,29 +529,42 @@ async function compileJsWithEsbuild(srcDir, srcFilename, destDir, options) {
   let result = null;
 
   /**
-   * @param {number} time
+   * @param {number} startTime
    * @return {Promise<void>}
    */
-  async function build(time) {
+  async function build(startTime) {
     if (!result) {
       result = await esbuild.build({
         entryPoints: [entryPoint],
         bundle: true,
         sourcemap: true,
         outfile: destFile,
         plugins,
-        minify: options.minify,
         format: options.outputFormat || undefined,
+        // For es5 builds, ensure esbuild-injected code is transpiled.
         target: argv.esm ? 'es6' : 'es5',
         incremental: !!options.watch,
         logLevel: 'silent',
         external: options.externalDependencies,
+        write: false,
       });
     } else {
       result = await result.rebuild();
     }
-    await minifyWithTerser(destDir, destFilename, options);
-    await finishBundle(srcFilename, destDir, destFilename, options, time);
+    let code = result.outputFiles.find(({path}) => !path.endsWith('.map')).text;
+    let map = result.outputFiles.find(({path}) => path.endsWith('.map')).text;
+
+    if (options.minify) {
+      ({code, map} = await minify(code, map));
+    }
+
+    await Promise.all([
+      fs.outputFile(destFile, code),
+      fs.outputFile(`${destFile}.map`, map),
+    ]);
+
+    // TODO: finishBundle should operate in-mem instead of reading/writing from FS.
```

Comment:
Agree with the sentiment behind this statement. In practice, the in-memory stuff we used to do with `gulp` streams once upon a time was way slower than the current direct-to-file-system operations. Going back to in-memory (without `gulp`) will be worth it if we can achieve yet another order of magnitude speed up.

Reply:
Thank you for the context! I know we'll need to leave some high level funcs that operate on FS around for test infra, but I can't imagine in-mem being slower for this flow